### DISCUSSION TITLE
How to recover deleted text messages on Android using ADB without root?

### DISCUSSION THREAD

**Discussion by shariqkhan12345** [community/community#46250] [GitHub Education]
Hi Everyone,

Hope all of you are doing well.

My name is Muhammad Shariq Khan, I am a professional content writer. I've been stuck on a topic from a couple of days as I am unable to find the solution.

The topic is regarding *How to retrieve/recover deleted text messages on Android without root using ADB (Android Debug Bridge).*

So could you please help me in this regard?

Thanks,

Muhammad Shariq Khan

**Comment by samus-aran** [comment]
Hi @shariqkhan12345 ,

Thanks for being a part of the GitHub Community, we're glad you're here!

If you're looking for help for this specific topic, you might want to try asking for help somewhere that focuses on this project, such as the [android developer community](https://developer.android.com/studio/command-line/adb). It's possible that another GitHub user might have run into this same issue and can help, but the GitHub Community on Discussions focuses primarily on topics related to GitHub itself or collaboration on project development and ideas. We want to make sure you’re getting the best support you can, but this space may not be the right place for this particular topic.

Best of luck!
